src/main/java/frc/robot/subsystems/climber/Climber.java

```diff
@@ -0,0 +1,231 @@
+import com.ctre.phoenix.motorcontrol.TalonFXInvertType;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.simulation.BatterySim;
+import edu.wpi.first.wpilibj.simulation.EncoderSim;
+import edu.wpi.first.wpilibj.simulation.RoboRioSim;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.Robot;
+import frc.robot.subsystems.UnitModel;
+
+
+public class Climber extends SubsystemBase {
+
+    private static Climber INSTANCE = null;
+
+    private final WPI_TalonFX auxMotor = new WPI_TalonFX(Ports.Climber.AUX);
+    private final WPI_TalonFX mainMotor = new WPI_TalonFX(Ports.Climber.MAIN_MOTOR);
+    private final Solenoid stopper = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Climber.STOPPER);
+    private final UnitModel unitModel = new UnitModel(Constants.Climber.TICKS_PER_RAD);
+
+    private final ArmFeedforward feedforward = new ArmFeedforward(Constants.Climber.F_FORWARD_S, Constants.Climber.F_FORWARD_COS, Constants.Climber.F_FORWARD_V, Constants.Climber.F_FORWARD_A);
+
+    private final PIDController controller = new PIDController(Constants.Climber.P_VELOCITY, Constants.Climber.I_VELOCITY, Constants.Climber.D_VELOCITY);
+
+    private final Encoder encoder = new Encoder(Ports.Climber.ENCODER_A_CHANNEL, Ports.Climber.ENCODER_B_CHANNEL);
+    private final DCMotor armGearbox = DCMotor.getFalcon500(2);
+
+    private final SingleJointedArmSim m_armSim =
+            new SingleJointedArmSim(
+                    armGearbox,
+                    Constants.Climber.GEAR_RATIO,
+                    SingleJointedArmSim.estimateMOI(Constants.Climber.ARM_LENGTH, Constants.Climber.ARM_MASS),
+                    Constants.Climber.ARM_LENGTH,
+                    Units.degreesToRadians(Constants.Climber.MIN_ANGLE),
+                    Units.degreesToRadians(Constants.Climber.MAX_ANGLE),
+                    Constants.Climber.ARM_MASS,
+                    true,
+                    VecBuilder.fill(Constants.Climber.ARM_ENCODER_DIST_PER_PULSE)
+            );
+
+    private final EncoderSim encoderSim = new EncoderSim(encoder);
```

Comment:
CTRE devices support simulation, do you need this encoder simulator too (really asking, I'm not familiar with simulations).

Reply:
We do because the TalonFXSimCollection does not have a get velocity method